scripts/build_podspecs.py

```diff
@@ -0,0 +1,183 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import random
+import string
+import argparse
+
+class Dependency:
+    def __init__(self, name, version="s.version.to_s", internal=True):
+        self.name = name
+        self.version = version
+        self.internal = internal
+
+    def as_podspec(self):
+        if self.internal:
+            return "    s.dependency '%s', %s\n" % (self.name, self.version)
+        else: 
+            return "    s.dependency '%s', '%s'\n" % (self.name, self.version)
+
+class Pod:
+    def __init__(self, name, module_name, version, dependencies=None):
+        self.name = name
+        self.module_name = module_name
+        self.version = version
+
+        if dependencies is None:
+            dependencies = []
+
+        self.dependencies = dependencies
+    
+    def add_dependency(self, dependency):
+        self.dependencies.append(dependency)
+    
+    def as_podspec(self):
+        print('\n')
+        print('Building Podspec for %s' % self.name)
+        print('-----------------------------------------------------------')
+
+        podspec = "Pod::Spec.new do |s|\n\n"
+        podspec += "    s.name = '%s'\n" % self.name
+        podspec += "    s.module_name = '%s'\n" % self.module_name
+        podspec += "    s.version = '%s'\n" % self.version
+        podspec += "    s.license = { :type => 'Apache 2.0', :file => 'LICENSE' }\n"
+        podspec += "    s.summary = 'Swift gRPC code generator plugin and runtime library'\n"
+        podspec += "    s.homepage = 'https://www.grpc.io'\n"
+        podspec += "    s.authors  = { 'The gRPC contributors' => 'grpc-packages@google.com' }\n\n"
+
+        podspec += "    s.source = { :git => 'https://github.com/grpc/grpc-swift.git', :tag => s.version }\n\n"
+
+        podspec += "    s.swift_version = '5.0'\n"
+
+        podspec += "    s.ios.deployment_target = '10.0'\n"
+        podspec += "    s.osx.deployment_target = '10.12'\n"
+        podspec += "    s.tvos.deployment_target = '10.0'\n"
+        
+        podspec += "    s.source_files = 'Sources/%s/**/*.{swift,c,h}'\n" % (self.module_name)
+
+        podspec += "\n" if len(self.dependencies) > 0 else ""
+
+        for dep in self.dependencies:
+            podspec += dep.as_podspec()
+
+        podspec += "\nend"
+        return podspec
+
+class PodManager:
+    pods = []
+
+    def __init__(self, directory, version, should_publish):
+        self.directory = directory
+        self.version = version
+        self.should_publish = should_publish
+
+    def write(self, pod, contents):
+        print("    Writing to %s/%s.podspec " % (self.directory, pod))
+        with open("%s/%s.podspec" % (self.directory, pod), "w") as f: 
+            f.write(contents)
+    
+    def publish(self, pod_name):
+        os.system('pod repo update')
```

Comment:
Is the mixing of single and double quotes intentional? (I'm not a Python expert.)

Reply:
Not quite lol It was mostly me just being inconsistent. Cleaned it up some! 👍